src/main/java/org/wildfly/security/credential/external/impl/CmdCredentialProvider.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.credential.external.impl;
+
+import static org.wildfly.security._private.ElytronMessages.log;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.security.AccessController;
+import java.security.PrivilegedActionException;
+import java.security.PrivilegedExceptionAction;
+
+import org.wildfly.security.credential.Credential;
+import org.wildfly.security.credential.external.ExternalCredentialException;
+import org.wildfly.security.credential.store.CredentialStorePermission;
+
+/**
+ * @author <a href="mailto:pskopek@redhat.com">Peter Skopek</a>
+ */
+public class CmdCredentialProvider extends ExecCredentialProvider {
```

Comment:
This would also work as a CallbackHandler, still just using the two basic callback types, NameCallback and CredentialCallback.